syntax = "proto2";

package addon;

// -- Add-on operations --------------------------------------------------------

message LoginRequest {
  required uint32 game_version_major = 1;
  required uint32 game_version_minor = 2;
  required uint32 game_version_point = 3;
}

message LoginResponse {
  required uint32 result = 1;
}

message LogoutRequest {
}

message LogoutResponse {
}

message GetStatusRequest {
}

message GetStatusResponse {
  required uint32 result = 1;
}

message AnnounceRequest {
  required string flag = 1;
  required string sender = 2;
  required string msg = 3;
}

message AnnounceResponse {
}

// --- Add-on callbacks --------------------------------------------------------

message LogRequest {
  required uint32 level = 1;
  required string msg = 2;
}

message LogResponse {
}

message GetSettingRequest {
  required string setting_name = 1;
  required string settings_value = 2;
}

message GetSettingResponse {
  required bool result = 1;
}

message QueueNotificationRequest {
  required uint32 type = 1;
  required string msg = 2;
}

message QueueNotificationResponse {
}

message WakeOnLanRequest {
  required string mac_address = 1;
}

message WakeOnLanResponse {
  required bool result = 1;
}

message UnknownToUTF8Request {
  required string str = 1;
}

message UnknownToUTF8Response {
  required string result = 1;
}

message GetLocalizedStringRequest {
  required uint32 code = 1;
}

message GetLocalizedStringResponse {
  required string result = 1;
}

message GetDVDMenuLanguageRequest {
}

message GetDVDMenuLanguageResponse {
  required string result = 1;
}

